compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -504,12 +558,14 @@ impl<'tcx> Stable<'tcx> for ty::GenericArgs<'tcx> {
     }
 }
 
-impl<'tcx> Stable<'tcx> for ty::PolyFnSig<'tcx> {
-    type T = stable_mir::ty::PolyFnSig;
-    fn stable(&self, tables: &mut Tables<'tcx>) -> Self::T {
-        use stable_mir::ty::Binder;
+impl<'tcx, S, T> Stable<'tcx> for ty::Binder<'tcx, S>
```

Comment:
Can you rename the T here to V. I think it will be clearer because we won't have the T param and the T associated type and won't have this T = T thing that looks weird :).